### ISSUE TITLE
Any idea how to improve the performance when handling large ontologies?

### ISSUE THREAD

**Issue by leonqli** [lambdamusic/Ontospy#44]

**Comment by lambdamusic** [comment]
Do you have any sample ontology in mind? I'm trying to wrap my head around this problem so it'd be useful to have some sample models for testing.

**Comment by leonqli** [comment]
you may try with this ontology: http://bioportal.bioontology.org/ontologies/NCBITAXON

**Comment by lambdamusic** [comment]
The main problem is that ontospy attempts to build the entire ontology model in memory, and that takes time if there are many classes and properties to correlated. 

I've tried using threads, but with no real performance improvements as the main tasks (extract classes, properties, concepts etc..) tend to be reliant on each other.

For very large ontologies maybe it's more indicated to use a triplestore. Otherwise I'm kind of out of ideas here..

**Comment by leonqli** [comment]
You may want to take look of https://pythonhosted.org/Owlready2/  It seems to having better performance on large ontologies.

**Comment by lambdamusic** [comment]
Thanks! Looks like they use an ad-hoc back end, maybe that's it. Will look more into it

**Comment by leonqli** [comment]
Yes, they use SQLite as backend. Do you think it is helpful for improving the performance of ontospy?

**Comment by jclerman** [comment]
It's also not too difficult to load an ontology into Apache Fuseki Jena.  The main issue is the non-Python dependency (Fuseki), but once the store is running it's easy to use rdflib to mediate querying.
